swe_roe/ex2b.c

```diff
@@ -1490,52 +1699,17 @@ PetscErrorCode RHSFunctionForBoundaryEdges(RDyApp app, Vec F, PetscReal *amax_va
     cn_vec_bnd[ii] = edges->cn[iedge];
     sn_vec_bnd[ii] = edges->sn[iedge];
 
-    PetscBool is_edge_vertical;
-    if (PetscAbs(edges->normals[iedge].V[0]) < 1.e-10) {
-      is_edge_vertical = PETSC_TRUE;
-    } else if (PetscAbs(edges->normals[iedge].V[1]) < 1.e-10) {
-      is_edge_vertical = PETSC_FALSE;
-    } else {
-      printf("The code only support quad cells with edges that align with x and y axis\n");
-      exit(0);
-    }
-
     if (cells->is_local[l] && b_ptr[l] == 0) {
       // Perform computation for a boundary edge
 
-      PetscBool bnd_cell_order_flipped = PETSC_FALSE;
-
-      if (is_edge_vertical) {
-        if (cells->centroids[l].X[1] > edges->centroids[iedge].X[1]) bnd_cell_order_flipped = PETSC_TRUE;
-      } else {
-        if (cells->centroids[l].X[0] > edges->centroids[iedge].X[0]) bnd_cell_order_flipped = PETSC_TRUE;
-      }
-
       if (cells->is_local[l] && b_ptr[l] == 0) {
         hr_vec_bnd[ii] = hl_vec_bnd[ii];
-        if (is_edge_vertical) {
-          ur_vec_bnd[ii] = ul_vec_bnd[ii];
-          vr_vec_bnd[ii] = -vl_vec_bnd[ii];
-        } else {
-          ur_vec_bnd[ii] = -ul_vec_bnd[ii];
-          vr_vec_bnd[ii] = vl_vec_bnd[ii];
-        }
-
-        if (bnd_cell_order_flipped) {
-          PetscReal tmp;
-
-          tmp            = hl_vec_bnd[ii];
-          hl_vec_bnd[ii] = hr_vec_bnd[ii];
-          hr_vec_bnd[ii] = tmp;
 
-          tmp            = ul_vec_bnd[ii];
-          ul_vec_bnd[ii] = ur_vec_bnd[ii];
-          ur_vec_bnd[ii] = tmp;
+        PetscReal dum1 = PetscPowReal(sn_vec_bnd[ii], 2.0) - PetscPowReal(cn_vec_bnd[ii], 2.0);
```

Comment:
Consider simple squares here too.

Reply:
There is `PetscPowRealInt` that does integer powers in a way the compiler can more clearly optimize. We have `Square(x)` helpers in most of my other projects because it's just simpler to read. I dislike typing that stuff out.

BTW, many compilers optimize small integer compile-time powers. NVRTC chokes on use of `pow(x, 2)` where the exponent is an integer.

Reply:
That's a good point. I've also used macros like

```
#define Square(x) ((x)*(x))
```

in these types of situations. @bishtgautam , it's up to you if you prefer that over writing out each term twice here. But in RDycore proper, we'll likely want something like this.

Reply:
I will add the macro in this PR so we don't forget about it.